OpenDreamClient/Interface/Descriptors/InterfaceDescriptor.cs

```diff
@@ -13,14 +13,25 @@ public sealed class InterfaceDescriptor {
         MacroSetDescriptors = macroSetDescriptors;
         MenuDescriptors = menuDescriptors;
     }
+
+    public InterfaceDescriptor Copy(ISerializationManager serializationManager) {
+        return new InterfaceDescriptor(serializationManager.CreateCopy(WindowDescriptors),
+            serializationManager.CreateCopy(MacroSetDescriptors), serializationManager.CreateCopy(MenuDescriptors));
+    }
 }
 
 [Virtual, ImplicitDataDefinitionForInheritors]
 public class ElementDescriptor {
     [DataField("type")]
-    public string Type;
+    public readonly string Type;
```

Comment:
The serialization manager is throwing an exception when attempting to copy descriptors because this is null. Macro sets and menus currently have no type, the correct fix would probably be to give them one.